gnucash/gnome/gnc-plugin-page-account-tree.c

```diff
@@ -1499,21 +1500,51 @@ gnc_plugin_page_account_tree_cmd_delete_account (GtkAction *action, GncPluginPag
          * Note that one effect of the modal dialog is preventing
          * the account selectors from being repopulated.
          */
-        response = gtk_dialog_run(GTK_DIALOG(dialog));
-        if (GTK_RESPONSE_ACCEPT != response)
+        account_currency = gnc_account_get_currency_or_parent (account);
+        // Ideally we should check all subaccount currencies, and verify they're all identical. If not, it's not advisable
+        // to move subaccount currencies to a new account.
+        while (1)
```

Comment:
use `TRUE` instead of `1`,  but better to use `!ta_match`. 

Plus this function is an appalling 350 lines with multiple unconditional blocks, so extracting this to a function would be even better.

Reply:
If I put this in a separate function, I'll have to pass 3 Account** parameters to the function for ta, saa and sta, or a structure, or a list. Worth it? I have no problem passing Account** params myself, but I've heard you object to it.

Reply:
@christopherlam doesn't like out-params because it's not "functional". I don't mind them, though I think that returning a struct is better. I wonder if gcc does RVO on C or only on C++...

Even better, you could call the functions to move the accounts and transactions from the function that runs the dialog and return a gboolean to indicate whether the account deletion should proceed.

Reply:
I object to mutating arguments only for functions that are expected to be used in guile reports :wink:  I have an ongoing fight with `gncOwner.c` as a result; not only many functions mutate the `gncOwner *` argument, they also force me to `gncOwnerFree` it. :vomiting_face: